client_files/ElasticSearch.sh

```diff
@@ -0,0 +1,167 @@
+#
+# This script installs everything required to use elasticsearch in MediaWiki
+# 
+# Dependencies
+# - PHP compiled with cURL
+# - Elasticsearch
+#   - JAVA 7+
+# - Extension:Elastica
+# - Extension:CirrusSearch
+# 
+# Ref:
+# https://www.mediawiki.org/wiki/Extension:CirrusSearch
+# https://en.wikipedia.org/w/api.php?action=cirrus-config-dump&srbackend=CirrusSearch&format=json
+# https://git.wikimedia.org/blob/mediawiki%2Fextensions%2FCirrusSearch.git/REL1_25/README
+# https://www.mediawiki.org/wiki/Extension:CirrusSearch/Tour
+# https://wikitech.wikimedia.org/wiki/Search
+#
+
+#
+# Install JAVA
+# 
+# http://docs.oracle.com/javase/8/docs/technotes/guides/install/linux_jdk.html#BJFJHFDD
+# http://stackoverflow.com/questions/10268583/how-to-automate-download-and-installation-of-java-jdk-on-linux
+#
+echo "******* Downloading and installing JAVA Development Kit *******"
+cd ~/sources/meza1/client_files
+yum -y install java-1.7.0-openjdk
+# Reference this for if we want to try JDK 8: http://tecadmin.net/install-java-8-on-centos-rhel-and-fedora/
+## wget --no-check-certificate --no-cookies --header "Cookie: oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jdk/8u45-b14/jdk-8u45-linux-x64.rpm
+## rpm -ivh jdk-8u45-linux-x64.rpm
+
+# Verify JAVA is installed
+java -version
+
+# Set $JAVA_HOME
+#
+# http://askubuntu.com/questions/175514/how-to-set-java-home-for-openjdk
+#
+echo "JAVA_HOME=\"/usr/bin\"" >> /etc/environment
+source /etc/environment
```

Comment:
The above method is for Debian/Ubuntu. Need to test if it is doing what is expected in CentOS/RedHat. I believe in CentOS/RedHat the preferred method is to do:

```bash
echo "export PATH=/usr/local/php/bin:\$PATH" > /etc/profile.d/php.sh
echo "export JAVA_HOME=/usr/bin" > /etc/profile.d/java.sh
source /etc/profile.d/java.sh
```

The one caveat to that is that all the troubles I was having with updating the PATH variable for PHP was that it would update the PATH within that particular script (so running ```php --version``` would work), but on a later script it wouldn't remain updated (so running ```php --version``` would not work). To get around this issue I did the PATH update in ```install.sh``` immediately following ```php.sh```, such that the update would carry over into the following ```mysql.sh``` (a child-process of ```install.sh```, which thus inherits the variables).

I'll run some tests to see which way we should do things.